src/factorization.jl

```diff
@@ -144,9 +144,42 @@ function restarted_arnoldi(A::AbstractMatrix{T}, min = 5, max = 30, tolerance =
     for restarts = 1 : max_restarts
         iterate_arnoldi!(A, arnoldi, min + 1 : max)
         implicit_restart!(arnoldi, min, max, active)
-        active = detect_convergence!(view(arnoldi.H, 1:min+1, 1:min), tolerance)
+
+        new_active = detect_convergence!(view(arnoldi.H, 1:min+1, 1:min), tolerance)
+
+        # Bring the new locked part oF H into upper triangular form
+        if new_active > active 
+            schur_form = schur(view(arnoldi.H, active : new_active - 1, active : new_active - 1))
+            arnoldi.H[active : new_active - 1, active : new_active - 1] = schur_form[1]
+
+            V_locked = view(arnoldi.V, :, active : new_active - 1)
+            H_right = view(arnoldi.H, active : new_active - 1, new_active : min)
+
+            A_mul_B!(V_locked, copy(V_locked), schur_form[2])
+            Ac_mul_B!(H_right, schur_form[2], copy(H_right))
+            
+            if active > 1 
+                H_above = view(arnoldi.H, 1 : active - 1, active : new_active - 1)
+                A_mul_B!(H_above, copy(H_above), schur_form[2])
+            end
+            
+            active = new_active
+        end
         @show active
     end
 
+    # At the end, bring the rest of H into upper triangular form as well
```

Comment:
This is something causing the tests to fail -- accuracy will be at most `abs(H[min + 1, min])`, and that number is not yet close to zero.

We have not completely decided on an API for the method, but let's suppose the user wants `k` eigenvectors, with `k <= min < max`. This value `k` then determines when we can exit the `for  restarts = 1 : max_restarts`-loop -- exactly when `new_active > k`, just break the loop after the transform to Schur form. 

Then we stil got to decide what to return, and I think what we should return is the _full_ Arnoldi decomposition + a number that tells how many vectors are locked, `active - 1` to be precise. Maybe we could just create a `struct PartialSchur{TQ,TR}; Q::TQ; R::TR; k::Int end` where `Q` just is `V` and `R` just `H` such that `A*Q[:, 1 : max] = Q * R` holds, and in particular, where `A * Q[:, 1 : k] = Q[:, 1 : k] * R[1 : k, 1 : k]` is a partial Schur decomposition :). Let's not return less cause the user might still be interested in the half-converged Arnoldi-decomposition.